app/controllers/albumsController.js

```diff
@@ -50,4 +57,122 @@ const showPhotoFromAlbumByIds = (req, res) => {
+const buyAlbum = async (req, res) => {
+  const { id } = req.params;
+
+  const token =
+    (req.body && req.body.access_token) ||
+    (req.query && req.query.access_token) ||
+    req.headers['x-access-token'];
+
+  const newUser = await obtainOneUser({ where: { email: verifyToken(token).email } });
+  const user = newUser.dataValues;
+
+  const album = await getAlbumById(id);
+
+  const userId = user.id;
+  const externalReferenceId = album.body.id;
+  const purchase = { userId, externalReferenceId };
+
+  const previousPurchase = await obtainOnePurchase({ where: { userId, externalReferenceId } });
+
+  if (previousPurchase) {
+    logger.info('The album was already purchased.');
+    return res.status(500).send('The album was already purchased.');
+  }
+
+  return registerPurchase(purchase)
+    .then(result => {
+      logger.info(
+        // eslint-disable-next-line max-len
+        `The user ${user.firstName} ${user.lastName} successfully purchased the album "${album.body.title}", id: ${id}, ${result}`
+      );
+      res.send(
+        `The user ${user.firstName} ${user.lastName} successfully purchased the album "${album.body.title}", id: ${id}`
+      );
+    })
+    .catch(error => {
+      logger.error(`There were errors during the purchase of album "${album.body.title}", id: ${id}`);
+      res.status(500).send({
+        error: `There were errors during the purchase of album "${
+          album.body.title
+        }", id: ${id} : ${JSON.stringify(error)}`
+      });
+    });
+};
+
+const listAlbumsFromUser = async (req, res) => {
+  const { user_id } = req.params;
+
+  const token =
+    (req.body && req.body.access_token) ||
+    (req.query && req.query.access_token) ||
+    req.headers['x-access-token'];
+
+  const newUser = await obtainOneUser({ where: { email: verifyToken(token).email } });
+  const user = newUser.dataValues;
+
+  if (user.privilegeLevel === 'normal' && user.id !== parseInt(user_id)) {
+    logger.info("This user cannot access these User's albums.");
+    return res.status(400).send("This user cannot access these User's albums.");
+  }
+
+  const purchases = await obtainAllPurchases({ where: { userId: user_id } });
+  const idsOfAlbums = purchases.map(element => element.dataValues.externalReferenceId);
+  const albumsFromUser = await Promise.all(idsOfAlbums.map(result => getAlbumById(result)));
+  const titlesFromAlbumsFromUser = albumsFromUser.map(result => result.body.title);
+
+  if (!purchases) {
+    logger.info('The albums could not be obtained');
+    return res.status(400).send('The albums could not be obtained');
+  }
+
+  logger.info(`The user with the id: ${user_id} has the following albums: ${titlesFromAlbumsFromUser}`);
+  return res
+    .status(200)
+    .send(`The user with the id: ${user_id} has the following albums: ${titlesFromAlbumsFromUser}`);
+};
+
+const listPhotosFromAlbum = async (req, res) => {
+  const { album_id } = req.params;
+
+  const token =
```

Comment:
I've seen this code more than once, so it would be nice to take it out and place it in a helper.